src/update-feeds/update-feeds.ts

```diff
@@ -124,8 +138,84 @@ export const runUpdateFeed = async (providerName: string, chain: Chain, chainId:
   });
 };
 
-// eslint-disable-next-line @typescript-eslint/require-await
-export const processBatch = async (batch: ReadDapiWithIndexResponse[]) => {
+export const mergeUrls = (receivedUrls: UrlSet[], freshExistingUrls: UrlSet[]) =>
+  Object.values(groupBy([...receivedUrls, ...freshExistingUrls], 'url')).flatMap(
+    (group) => sortBy(group, 'lastReceivedMs').pop()!
+  );
+
+export const updateDynamicState = (batch: ReadDapiWithIndexResponsesAndChainId) => {
+  batch.map((item) =>
+    updateState((draft) => {
+      const receivedUrls = item.signedApiUrls.flatMap((url) =>
+        item.dataFeed.dataFeeds.flatMap((dataFeed) => ({
+          url: `${url}/${dataFeed.airnodeAddress}`,
+          lastReceivedMs: Date.now(),
+        }))
+      );
+
+      const freshExistingUrls = draft.signedApiUrlStore.filter(
+        (url) => Date.now() - url.lastReceivedMs > 1000 * 60 * SIGNED_URL_EXPIRY_IN_MINUTES
+      );
+
+      // Appends records that don't already exist, updates records that already exist with new timestamps
+      draft.signedApiUrlStore = mergeUrls(receivedUrls.flat(), freshExistingUrls);
+
+      const cachedDapiResponse = draft.dapis[item.dapiName];
+
+      draft.dapis[item.dapiName] = {
+        dataFeed: cachedDapiResponse?.dataFeed ?? item.dataFeed,
+        dataFeedValues: { ...cachedDapiResponse?.dataFeedValues, [item.chainId]: item.dataFeedValue },
+        updateParameters: { ...cachedDapiResponse?.updateParameters, [item.chainId]: item.updateParameters },
+      };
+    })
+  );
+};
+
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId) =>
+  batch
+    .map((dapiResponse) => {
```

Comment:
This returns 3 possible values:
1. dapi response + shouldUpdate=false
2. dapi response + signed data
3. false

It would be simpler if you do a map where you return the signed data (which can be undefined) for each data feed and then use `filter` to filter out those that we don't want to update.